server/src/com/thoughtworks/go/server/dao/JobInstanceSqlMapDao.java

```diff
@@ -272,6 +272,11 @@ public int totalCompletedJobsOnAgent(String uuid) {
         return (Integer) getSqlMapClientTemplate().queryForObject("totalCompletedJobsOnAgent", arguments("uuid", uuid).asMap());
     }
 
+    @Override
+    public boolean isJobCompleted(JobIdentifier jobIdentifier) {
+        return mostRecentJobWithTransitions(jobIdentifier).isCompleted();
```

Comment:
I think we can use the latestCompletedJobs, restrict the count to 1 instead of querying for transitions. Since we have the job identifier, we can get the pipeline name, stage name, job name that's required for the latestCompletedJobs method? What do you think?

Reply:
@varshavaradarajan -- `latestCompletedJobs` for the given identifier can't be used.
Let's say for a job identifier, build 1 passed and build 2 is in progress, in such scenarios,

`latestCompletedJobs` will return `build1`, whereas,
`mostRecentJobWithTransitions` will return `build2`